python-package/lightgbm/sklearn.py

```diff
@@ -862,12 +906,29 @@ def fit(
         params["metric"] = [metric for metric in params["metric"] if metric is not None]
 
         if not isinstance(X, (pd_DataFrame, dt_DataTable)):
-            _X, _y = _LGBMCheckXY(X, y, accept_sparse=True, force_all_finite=False, ensure_min_samples=2)
+            _X, _y = _LGBMValidateData(
+                self,
+                X,
+                y,
+                # Prevent scikit-learn from deleting or modifying attributes like 'feature_names_in_' and 'n_features_in_'.
+                # We prefer to expose these via @property, to be able to raise a NotFittedError if they're accessed on an
+                # unfitted model... and so don't want to take on the complexity of defining setters and deleters for those.
```

Comment:
Not actual after the latest refactoring.
```suggestion
```

Reply:
Ah you are right! Thank you for catching this.